src/python/torchdistx/optimizers/anyprecision_optimizer.py

```diff
@@ -8,15 +8,32 @@
 # with optional Kahan summation for high precision weight updates.
 # Allows direct control over momentum, variance and auxiliary compensation
 # buffer dtypes.
-# Optional Kahan summation is used to offset precision reduction for
-# the weight updates. This allows full training in BFloat16 (equal or
-# better than FP32 results in many cases) due to high precision weight upates.
+# Optional Kahan summation is used to enable high precision for
+# the weight updates. This allows sucessful training in pure BFloat16
+# (often equal or better than FP32 results) due to high precision weight
+# updates, while training with reduced GPU memory and
+# increased training speed.
 
 import torch
+import torch.cuda.nccl as nccl
+import torch.distributed as dist
+from pkg_resources import packaging
```

Comment:
Looks like this doesn't include by default and I'm always hesitant to add a user dependency. Could we manually parse the string?

Reply:
agreed.  I've rewritten to do our own parse and removed the import.  
Fortunately we only require >= 11.0, which means only checking the major version:
~~~
cuda_version_major, _ = torch.version.cuda.split(".", maxsplit=1)
~~~
I did the maxsplit 1 to avoid any unpacking error from cuda being longer say 11.6.01 etc.  in the future.